src/app.ts

```diff
@@ -1,13 +1,24 @@
 import express from 'express'
 import path from 'path'
+import cookieParser from 'cookie-parser'
 const app = express()
 const port = process.env.PORT || 3000
+app.use(express.urlencoded())
+app.use(cookieParser())
 app.set('view engine', 'ejs')
 app.set('views', path.join(__dirname, '../../src/views'))
 
+app.get('/login', (req, res) => res.render('login', { error: '' }))
+app.post('/login', (req, res) => {
+  if (process.env.ADMIN !== req.body.adminPass)
+    return res.render('login', { error: 'Wrong Admin Password' })
+  res.cookie('adminPass', req.body.adminPass)
+  res.redirect('/admin/serviceHostKeys')
+})
 app.get('/', (req, res) => res.render('index', { message: 'Hello myProxy' }))
 
 app.get('/admin/serviceHostKeys', (req, res) => {
```

Comment:
You will need a check for all admin routes, so you need to add a middleware so all requests going through admin will get a check. Sample:

```
app.use('/admin', (req, res, next) => {
  if (!req.cookies.adminPass) return res.redirect('/login')
  return next()
})
```

Running the next function will continue the request to other handlers.